src/ResponseSerializerInterface.php

```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace Yiisoft\Yii\Rest;
+
+use Psr\Http\Message\ResponseInterface;
+
+interface ResponseSerializerInterface
+{
+    // TODO may be $code is redundant?
```

Comment:
I don't thin is it redundant.

Reply:
It can be set later after processing

Reply:
But it could be part of body. I've seen that multiple times in API requirements i.e.

```json
{
    status: 400,
    data: {
        message: "Bla-bla is not valid!"
    }
}
```

Reply:
No, it's wrong. 
`status: 400` is part of response, I mean that should be passed to the `serialize` method, eg. 
```php
$responseSerializer->serialize(200, ['status' => 400, 'data' => [/* ... */]]);
```

The first parameter `$code` is only HTTP status that can be replaced with https://github.com/php-fig/http-message/blob/master/src/ResponseInterface.php#L52 method

Reply:
OK. Then it's fine to remove it.